monitor/metric.go

```diff
@@ -22,102 +27,117 @@ const (
 	LatencyConnectorKey           = "latencyconn"
 	LatencyKafkaProduceKey        = "latencykafkaprod"
 	LatencyStreamserverConsumeKey = "latencyssconsume"
-
-	// Metric keys
-	rpcLatency    = 0
-	connLatency   = 1
-	coreLatency   = 2
-	systemLatency = 3
-	e2eLatency    = 4
 )
 
-var (
-	latencyObservations = map[string]int64{
-		LatencyRpcKey:                 0,
-		LatencyConnectorKey:           0,
-		LatencyKafkaProduceKey:        0,
-		LatencyStreamserverConsumeKey: 0,
-	}
-	latencyMetrics = map[int]int64{
-		rpcLatency:    0,
-		connLatency:   0,
-		coreLatency:   0,
-		systemLatency: 0,
-		e2eLatency:    0,
-	}
-	histograms = map[int]Histogram{
-		rpcLatency:    {name: "latency.rpc.histogram", description: "Latency from block time to connector reception"},
-		connLatency:   {name: "latency.connector.histogram", description: "Latency from connector reception to kafka produce"},
-		coreLatency:   {name: "latency.core.histogram", description: "Latency from kafka to streamserver"},
-		systemLatency: {name: "latency.system.histogram", description: "Latency from connector to streamserver"},
-		e2eLatency:    {name: "latency.e2e.histogram", description: "Latency from block time to streamserver"},
+// ExportLatencyMetrics exports all latency metrics from baggage in ctx
+func ExportLatencyMetrics(ctx context.Context, meter metric.Meter, connectorNmae string, env string) {
+	latency := latencyHistogram{
+		meter:         meter,
+		connectorName: connectorName,
+		env:           env,
+		value:         0,
 	}
-)
 
-// ExportLatencyMetrics exports latency metrics from baggage in ctx
-func ExportLatencyMetrics(ctx context.Context, meter metric.Meter, connName string, env string) {
-	bag := baggage.FromContext(ctx)
+	recordRpcLatency(ctx, latency)
```

Comment:
Even though these calls are `pass by reference` and should work fine, it is confusing to read as they seem to cause `side effects`. We can make them `pure functions`.
I see that these helper functions are not using any info from the `latency` object. Also, `recordLatencyHistogram(ctx, latency)` calls seem repetitive. 
Can you refactor these calls so that they return a specific `latency` object which will later be used by this driver function to call `recordLatencyHistogram(ctx, latency)`?
Something like this maybe:
```
for _, latency := range []Latency {
 getRPCLatency(),
getConnectorLatency(),
getSystemLatency(),
getEndLatency(),
}{
recordLatencyHistogram(ctx, latency)
}
```

Reply:
updated with your suggestions 👍